lib/ansible/module_utils/network/nxos/config/lldp_global/lldp_global.py

```diff
@@ -0,0 +1,244 @@
+            return []
+        return lldp_global_facts
+
+    def execute_module(self):
+        """ Execute the module
+
+        :rtype: A dictionary
+        :returns: The result from module execution
+        """
+        result = {'changed': False}
+        commands = list()
+        warnings = list()
+
+        existing_lldp_global_facts = self.get_lldp_global_facts()
+        commands.extend(self.set_config(existing_lldp_global_facts))
+        if commands:
+            if not self._module.check_mode:
+                self._connection.edit_config(commands)
+            result['changed'] = True
+        result['commands'] = commands
+
+        changed_lldp_global_facts = self.get_lldp_global_facts()
+
+        result['before'] = dict(existing_lldp_global_facts)
+        if result['changed']:
+            result['after'] = dict(changed_lldp_global_facts)
+
+        result['warnings'] = warnings
+        return result
+
+    def set_config(self, existing_lldp_global_facts):
+        """ Collect the configuration from the args passed to the module,
+            collect the current configuration (as a dict from facts)
+
+        :rtype: A list
+        :returns: the commands necessary to migrate the current configuration
+                  to the desired configuration
+        """
+        want = self._module.params['config']
+        have = existing_lldp_global_facts
+        if len(have) == 0:
+            have = {}
+        resp = self.set_state(remove_empties(want), have)
+        return resp.split('\n')[:-1]
+
+    def set_state(self, want, have):
+        """ Select the appropriate function based on the state provided
+
+        :param want: the desired configuration as a dictionary
+        :param have: the current configuration as a dictionary
+        :rtype: A list
+        :returns: the commands necessary to migrate the current configuration
+                  to the desired configuration
+        """
+        state = self._module.params['state']
+        if state == 'deleted':
+            commands = self._state_deleted(have)
+        elif state == 'merged':
+            commands = self._state_merged(want, have)
+        elif state == 'replaced':
+            commands = self._state_replaced(want, have)
+        return commands
+
+    def _state_replaced(self, want, have):
+        """ The command generator when state is replaced
+
+        :rtype: A list
+        :returns: the commands necessary to migrate the current configuration
+                  to the desired configuration
+        """
+        commands = ''
+        diff = dict_diff(have, want)  # diff will contain unique and new values in want
+        wantk = want.keys()  # find keys in want to ignore them for defaults
+        havek = have.keys()
+        wantk = self.get_nested_keys(want, wantk)
+        havek = self.get_nested_keys(have, havek)
+        include_params = list(set(havek) - set(wantk))  # keys in have and not in want; they need to be defaulted
```

Comment:
Write a method to generate final diff so it can be used anywhere in the code when required.